ife-cognito/main.tf

```diff
@@ -83,4 +87,23 @@ resource "aws_cognito_user_pool_client" "client" {
   allowed_oauth_scopes                 = each.value.allowed_scopes
 
   depends_on = [aws_cognito_resource_server.ife_resource_server]
+}
+
+resource "random_password" "password" {
+  for_each = local.basic_auth_client_settings
+  length   = 16
+  special  = false
+  keepers = {
+    client = each.key
+  }
+}
+
+resource "aws_ssm_parameter" "basic_auth_client" {
+  for_each = local.basic_auth_client_settings
+
+  name  = "/${var.param_store_client_prefix}/client/${each.key}"
+  value = "{\"password\":${jsonencode(random_password.password[each.key].result)}, \"allowed_scopes\":\"${join(" ", each.value.allowed_scopes)}\"}"
```

Comment:
I am wondering if there is a possibility to read the scopes directly from Cognito and not store it here.

Reply:
I am still not sure if it makes sense to have an option to have basic auth without OAuth. If yes, then we need to have a place where we can keep client config without cognito. It means also scopes. Lets see what future use cases will be.